File: kokain-core-api/src/main/java/com/schwarz/kokain/api/EFactory.java
package com.schwarz.kokain.api;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import kotlin.reflect.KClass;


@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface EFactory {


    Class<?>[] additionalFactories() default Void.class;

}

File: kokain-core-api/src/main/java/com/schwarz/kokain/api/EFactory.kt
package com.schwarz.kokain.api

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class EFactory(val additionalFactories: Array<KClass<*>> = [Void::class])
